Errors in dynamic imports are not properly caught

Runtime
```
debian 10
deno 1.6.2
v8 8.8.278.2
typescript 4.1.3
```
------

```ts
//a.ts
import("./b.ts").catch(e=>{
  console.log("caught import b.ts error")
  console.error(e)
})

import("./c.ts").catch(e=>{
  console.log("caught import c.ts error")
  console.error(e)
})

setInterval(function () {
  import("./c.ts").catch((e) => {
    console.log("caught interval import c.ts error");
    console.error(e);
  });
}, 1e3);
```
```ts
//b.ts
// @ts-ignore
import "./b1.ts";
export default () => "error";
```
```ts
//c.ts
// @ts-ignore
await import("./c1.ts");
export default () => "error";
```

create those files and then run `deno run -A a.ts`, you will get log like below:
```
caught import b.ts error
TypeError: Cannot resolve module "file://yourdir/b1.ts" from "file://yourdir/b.ts".
    at file://yourdir/b.ts:3:0
caught import c.ts error
TypeError: Cannot resolve module "file://yourdir/c1.ts".
error: Uncaught TypeError: Cannot resolve module "file://yourdir/c1.ts".
```

`import('./c.ts')` first error is caught, but then second import is not caught and it make my program exit

```
~/dev/deno  master
▶ deno run --allow-read=. a.mjs
error: Cannot resolve module "file:///Users/biwanczuk/dev/deno/c1.mjs" from "file:///Users/biwanczuk/dev/deno/c.mjs".
    at file:///Users/biwanczuk/dev/deno/c.mjs:3:6

~/dev/deno  master
▶ node --harmony-top-level-await a.mjs
caught import b.mjs error
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/biwanczuk/dev/deno/b1.mjs' imported from /Users/biwanczuk/dev/deno/b.mjs
    at finalizeResolution (internal/modules/esm/resolve.js:276:11)
    at moduleResolve (internal/modules/esm/resolve.js:699:10)
    at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:810:11)
    at Loader.resolve (internal/modules/esm/loader.js:85:40)
    at Loader.getModuleJob (internal/modules/esm/loader.js:229:28)
    at ModuleWrap.<anonymous> (internal/modules/esm/module_job.js:51:40)
    at link (internal/modules/esm/module_job.js:50:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
caught import c.mjs error
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/biwanczuk/dev/deno/c1.mjs' imported from /Users/biwanczuk/dev/deno/c.mjs
    at finalizeResolution (internal/modules/esm/resolve.js:276:11)
    at moduleResolve (internal/modules/esm/resolve.js:699:10)
    at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:810:11)
    at Loader.resolve (internal/modules/esm/loader.js:85:40)
    at Loader.getModuleJob (internal/modules/esm/loader.js:229:28)
    at Loader.import (internal/modules/esm/loader.js:164:28)
    at importModuleDynamically (internal/modules/esm/translators.js:99:35)
    at exports.importModuleDynamicallyCallback (internal/process/esm_loader.js:30:14)
    at file:///Users/biwanczuk/dev/deno/c.mjs:3:1
    at ModuleJob.run (internal/modules/esm/module_job.js:146:23) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

Looks like a problem with catching errors in dynamic imports. I will look into that

--- a/cli/tests/085_dynamic_import_async_error.ts
+++ b/cli/tests/085_dynamic_import_async_error.ts
@@ -1,0 +1,5 @@
+try {
+  await import("./delayed_error.ts");
+} catch (error) {
+  console.log(`Caught: ${error.stack}`);
+}

--- a/cli/tests/085_dynamic_import_async_error.ts.out
+++ b/cli/tests/085_dynamic_import_async_error.ts.out
@@ -1,0 +1,2 @@
+[WILDCARD]Caught: Error: foo
+    at [WILDCARD]/delayed_error.ts:[WILDCARD]

--- a/cli/tests/delayed_error.ts
+++ b/cli/tests/delayed_error.ts
@@ -1,0 +1,2 @@
+await new Promise((r) => setTimeout(r, 100));
+throw new Error("foo");

--- a/cli/tests/integration_tests.rs
+++ b/cli/tests/integration_tests.rs
@@ -2779,6 +2779,11 @@ console.log("finish");
     output: "084_worker_custom_inspect.ts.out",
   });
 
+  itest!(_085_dynamic_import_async_error {
+    args: "run --allow-read 085_dynamic_import_async_error.ts",
+    output: "085_dynamic_import_async_error.ts.out",
+  });
+
   itest!(js_import_detect {
     args: "run --quiet --reload js_import_detect.ts",
     output: "js_import_detect.ts.out",

--- a/core/runtime.rs
+++ b/core/runtime.rs
@@ -822,6 +822,12 @@ impl JsRuntime {
         );
         let promise = v8::Local::<v8::Promise>::try_from(value)
           .expect("Expected to get promise as module evaluation result");
+        let empty_fn = |_scope: &mut v8::HandleScope,
+                        _args: v8::FunctionCallbackArguments,
+                        _rv: v8::ReturnValue| {};
+        let empty_fn = v8::FunctionTemplate::new(scope, empty_fn);
+        let empty_fn = empty_fn.get_function(scope).unwrap();
+        promise.catch(scope, empty_fn);
         let promise_global = v8::Global::new(scope, promise);
         let mut state = state_rc.borrow_mut();
         state.pending_promise_exceptions.remove(&promise_global);

